src/Perlang.Tests.Integration/LogicalOperator/And.cs

```diff
@@ -0,0 +1,159 @@
+using System.Linq;
+using Xunit;
+using static Perlang.Tests.Integration.EvalHelper;
+
+namespace Perlang.Tests.Integration.LogicalOperator
+{
+    // Tests based on Lox test suite:
+    // https://github.com/munificent/craftinginterpreters/blob/master/test/logical_operator/and.lox
+    // https://github.com/munificent/craftinginterpreters/blob/master/test/logical_operator/and_truth.lox
+    public class And
+    {
+        [Fact]
+        public void returns_the_first_falsy_argument_false_and_1()
+        {
+            string source = @"
+                print false and 1;
+            ";
+
+            string output = EvalReturningOutput(source).SingleOrDefault();
+
+            Assert.Equal("False", output);
+        }
+
+        [Fact]
+        public void returns_the_first_falsy_argument_true_and_1()
+        {
+            string source = @"
+                print true and 1;
+            ";
+
+            string output = EvalReturningOutput(source).SingleOrDefault();
+
+            Assert.Equal("1", output);
+        }
+
+        [Fact]
+        public void returns_the_first_falsy_argument_1_and_2_and_false()
+        {
+            string source = @"
+                print 1 and 2 and false;
+            ";
+
+            string output = EvalReturningOutput(source).SingleOrDefault();
+
+            Assert.Equal("False", output);
+        }
+
+        [Fact]
+        public void returns_the_last_argument_if_all_are_truthy_1_and_true()
+        {
+            string source = @"
+                print 1 and true;
+            ";
+
+            string output = EvalReturningOutput(source).SingleOrDefault();
+
+            Assert.Equal("True", output);
+        }
+
+        [Fact]
+        public void returns_the_last_argument_if_all_are_truthy_1_and_2_and_3()
+        {
+            string source = @"
+                print 1 and 2 and 3;
+            ";
+
+            string output = EvalReturningOutput(source).SingleOrDefault();
+
+            Assert.Equal("3", output);
+        }
```

Comment:
This whole "returns the last argument" stuff is just sketchy all the way from top to bottom. It may fit in  dynamic languages like Lox or Ruby (the latter has the similar behavior _but only with `&&`_, not with its different-precedence-operator named `and` :see_no_evil: :hear_no_evil: :speak_no_evil:) but it definitely feels odd in a Perlang context. We should revamp this before we consider #245 "done".